TestModels/SimpleTypes/SimpleEnum/codegen-patches/rust/dafny-4.5.0.patch

```diff
@@ -0,0 +1,2598 @@
+index d0133896..03d5c0e9 100644
+--- b/TestModels/SimpleTypes/SimpleEnum/runtimes/rust/dafny_impl/src/implementation_from_dafny.rs
++++ a/TestModels/SimpleTypes/SimpleEnum/runtimes/rust/dafny_impl/src/implementation_from_dafny.rs
+@@ -103,26 +103,12 @@ pub mod r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes {
+     #[derive(PartialEq, Clone)]
+     pub enum GetEnumInput {
+         GetEnumInput {
+-            value: ::std::rc::Rc<
+-                super::r#_Wrappers_Compile::Option<
+-                    ::std::rc::Rc<
+-                        super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape,
+-                    >,
+-                >,
+-            >,
++            value: ::std::rc::Rc<super::r#_Wrappers_Compile::Option<super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape>>,
+         },
+     }
+ 
+     impl GetEnumInput {
+-        pub fn value(
+-            &self,
+-        ) -> &::std::rc::Rc<
+-            super::r#_Wrappers_Compile::Option<
+-                ::std::rc::Rc<
+-                    super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape,
+-                >,
+-            >,
+-        > {
++        pub fn value(&self) -> &::std::rc::Rc<super::r#_Wrappers_Compile::Option<super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape>> {
+             match self {
+                 GetEnumInput::GetEnumInput { value } => value,
+             }
+@@ -179,26 +165,12 @@ pub mod r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes {
+     #[derive(PartialEq, Clone)]
+     pub enum GetEnumOutput {
+         GetEnumOutput {
+-            value: ::std::rc::Rc<
+-                super::r#_Wrappers_Compile::Option<
+-                    ::std::rc::Rc<
+-                        super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape,
+-                    >,
+-                >,
+-            >,
++            value: ::std::rc::Rc<super::r#_Wrappers_Compile::Option<super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape>>,
+         },
+     }
+ 
+     impl GetEnumOutput {
+-        pub fn value(
+-            &self,
+-        ) -> &::std::rc::Rc<
+-            super::r#_Wrappers_Compile::Option<
+-                ::std::rc::Rc<
+-                    super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape,
+-                >,
+-            >,
+-        > {
++        pub fn value(&self) -> &::std::rc::Rc<super::r#_Wrappers_Compile::Option<super::r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes::SimpleEnumShape>> {
+             match self {
+                 GetEnumOutput::GetEnumOutput { value } => value,
+             }
+@@ -304,9 +276,9 @@ pub mod r#_simple_dtypes_dsmithyenum_dinternaldafny_dtypes {
+ 
+     #[derive(PartialEq, Clone)]
+     pub enum SimpleEnumShape {
+-        FIRST {},
+-        SECOND {},
+-        THIRD {},
++        FIRST,
++        SECOND,
++        THIRD,
```

Comment:
You know what, let's NOT format `implementation_from_dafny.rs` for now, because that just creates a ton of patch noise

Reply:
But I didn't. I just took off the `{}` because it makes it much harder to deal with, and that's whatb the compiler ought to produce.

Reply:
Closing the loop for posterity: after offline conversation we determined this isn't a case where it's easy for the Dafny code generator to produce the most idiomatic code, and it's not customer-facing, so we're not patching the output after all.